### PR TITLE
About: Updating Hiring Section

### LINKED ISSUE #704: Update About UI
### Details:
Let's match iOS's About UI, so that Work with Us section looks great!

### PR DESCRIPTION
**Pull request by jleandroperez**
### Fix
In this PR we're updating the About UI, so that it matches iOS.

@eshurakov Incoming Quick PR! Thank youuuu!!
Closes #704 

### Test
1. Launch Simplenote
2. Click over the `Simplenote > About Simplenote` macOS Systembar Item

- [x] Verify the last section reads as `Work with Us` + `Are you a developer? We're hiring`
- [x] Verify that clicking over the `Work with Us` row opens the Hiring Website

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/About.storyboard [hunk 1]
```diff
@@ -157,110 +157,119 @@
...
+                                                        <color key="textColor" red="0.99999600649999998" green="1" blue="1" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
+                                                        <color key="backgroundColor" name="controlColor" catalog="System" colorSpace="catalog"/>
+                                                    </textFieldCell>
+                                                </textField>
+                                                <textField horizontalHuggingPriority="251" verticalHuggingPriority="750" fixedFrame="YES" setsMaxLayoutWidthAtFirstLayout="YES" translatesAutoresizingMaskIntoConstraints="NO" id="ZPK-KY-BDN">
+                                                    <rect key="frame" x="-2" y="15" width="76" height="19"/>
                                                     <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMinY="YES"/>
-                                                    <textFieldCell key="cell" sendsActionOnEndEditing="YES" title="Made with love by the folks at Automattic. Are you a developer? We're hiring." id="WV3-qs-WX7">
```
Comment (eshurakov) [inline Simplenote/About.storyboard:215]: No more love? 😢

**Review by eshurakov — APPROVED** [review]
Looks great! Tempting to apply 🎉 
:shipit:

File: Simplenote/About.storyboard (hunk 1 above)
Comment (jleandroperez) [reply Simplenote/About.storyboard:215]: LOL

**Comment by jleandroperez** [comment]
Thank you Evgeny!!

**Pull request merged**